ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthEstimateGas.java

```diff
@@ -85,19 +88,38 @@ private JsonCallParameter overrideGasLimitAndPrice(
   }
 
   private Function<TransactionSimulatorResult, JsonRpcResponse> gasEstimateResponse(
-      final JsonRpcRequestContext request) {
+      final JsonRpcRequestContext request, final EstimateGasOperationTracer operationTracer) {
     return result ->
         result.isSuccessful()
             ? new JsonRpcSuccessResponse(
-                request.getRequest().getId(), Quantity.create(result.getGasEstimate()))
+                request.getRequest().getId(),
+                Quantity.create(processEstimateGas(result, operationTracer)))
             : errorResponse(request, result.getValidationResult());
   }
 
+  /**
+   * Estimate gas by adding minimum gas remaining for some operation and the necessary gas for sub
+   * calls
+   *
+   * @param result transaction simulator result
+   * @param operationTracer estimate gas operation tracer
+   * @return estimate gas
+   */
+  private long processEstimateGas(
+      final TransactionSimulatorResult result, final EstimateGasOperationTracer operationTracer) {
+    // no more than 63/64s of the remaining gas can be passed to the sub calls
+    final double subCallMultiplier = Math.pow(65D / 64D, operationTracer.getMaxDepth());
```

Comment:
I would suggest to compute the 65/64 factor once and set it as a constant in this class.

Reply:
Done